### PR TITLE
Feature: Traefik v2 migration

### PR DESCRIPTION
**Pull request by vshatravenko**

### CODE REVIEW

File: templates/compose/gateway.yaml.erb [hunk 1]
```diff
@@ -8,10 +8,11 @@ services:
...
     command: /usr/local/bin/envoy -l info -c /etc/envoy/envoy.yaml
     labels:
-      traefik.enable: true
-      traefik.frontend.rule: "PathPrefix:/api/, /admin, /assets/;Host:<%= @config['app']['subdomain'] %>.<%= @config['app']['domain'] %>"
-      traefik.default.port: 8099
-      traefik.default.protocol: http
-      traefik.wss.frontend.rule: "Host:<%= @config['app']['subdomain'] %>.<%= @config['app']['domain'] %>"
-      traefik.wss.protocol: ws
-      traefik.wss.port: 8099
+      - "traefik.http.routers.gateway.rule=Host(`<%= @config['app']['subdomain'] %>.<%= @config['app']['domain'] %>`) && PathPrefix(`/api`,`/admin`,`/assets/`)"
```
Comment (vshatravenko) [inline templates/compose/gateway.yaml.erb]: I think /api, /assets and /admin are missing here

Comment (dpatsora) [reply templates/compose/gateway.yaml.erb]: No

Comment (dpatsora) [reply templates/compose/gateway.yaml.erb]: It works fine

File: templates/compose/proxy.yaml.erb [hunk 2]
```diff
@@ -3,13 +3,26 @@ version: '3.6'
...
+      - "--log.level=DEBUG"
+      - "--api.insecure=true"
+      - "--providers.docker=true"
+      - "--providers.docker.exposedbydefault=false"
+      - "--entrypoints.web.address=:80"
+      - "--entrypoints.websecure.address=:443"
+      <%- if @config['ssl']['enabled'] -%>
+      - "--certificatesresolvers.myresolver.acme.httpchallenge=true"
+      - "--certificatesresolvers.myresolver.acme.httpchallenge.entrypoint=web"
+      #- "--certificatesresolvers.myresolver.acme.caserver=https://acme-staging-v02.api.letsencrypt.org/directory"
```
Comment (vshatravenko) [inline templates/compose/proxy.yaml.erb]: Let's remove this line

Comment (dpatsora) [reply templates/compose/proxy.yaml.erb]: @akulakovaa fix it, please

Comment (akulakovaa) [reply templates/compose/proxy.yaml.erb]: okay

File: templates/compose/components-docs.yaml.erb [hunk 3]
```diff
@@ -6,7 +6,13 @@ services:
     expose:
       - "80"
     labels:
-      traefik.enable: true
-      traefik.frontend.rule: 'PathPrefix: /components, /src;Host: <%= @config['app']['subdomain'] %>.<%= @config['app']['domain'] %>'
-      traefik.port: 80
-      traefik.frontend.auth.basic: 'ILoveOpenware:$$apr1$$F47ty7.m$$PmAZpvByLBNgxNld4m1B81'
\ No newline at end of file
+      - "traefik.http.routers.frontend.rule=Host(`<%= @config['app']['subdomain'] %>.<%= @config['app']['domain'] %>`) && PathPrefix(`/components, /src`)"
```
Comment (dpatsora) [inline templates/compose/components-docs.yaml.erb]: traefik.http.routers.frontend-<%= @config['app']['name'] %>.rule=Host(`<%= @config['app']['subdomain'] %>.<%= @config['app']['domain'] %>`) && PathPrefix(`/components, /src`)

Comment (dpatsora) [reply templates/compose/components-docs.yaml.erb]: <%-  @name =  @config['app']['name'] -%>

traefik.http.routers.frontend-<%= @name %>.rule

File: templates/compose/components-docs.yaml.erb [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
     image: "<%= @config['images']['components-docs'] %>"
     expose:
       - "80"
     labels:
-      traefik.enable: true
-      traefik.frontend.rule: 'PathPrefix: /components, /src;Host: <%= @config['app']['subdomain'] %>.<%= @config['app']['domain'] %>'
-      traefik.port: 80
-      traefik.frontend.auth.basic: 'ILoveOpenware:$$apr1$$F47ty7.m$$PmAZpvByLBNgxNld4m1B81'
\ No newline at end of file
+      - "traefik.http.routers.frontend-<%= @name %>.rule=Host(`<%= @config['app']['subdomain'] %>.<%= @config['app']['domain'] %>`) && PathPrefix(`/components`, `/src`)"
```
Comment (dpatsora) [inline templates/compose/components-docs.yaml.erb]: change `frontend` to `components` in this file

File: templates/compose/superset.yaml.erb [hunk 5]
```diff
@@ -1,15 +1,22 @@
...
     image: "<%= @utils['images']['superset'] %>"
     restart: always
     volumes:
       - superset_db:/var/lib/superset
     labels:
-      traefik.enable: true
-      traefik.frontend.rule: "Host:superset.<%= @config['app']['domain'] %>"
-      traefik.port: 8088
+      - "traefik.http.routers.superset-<%= @name %>.rule=Host(`superset.<%= @config['app']['domain'] %>`)"
+      - "traefik.http.services.superset.loqadbalancer.server.port=8088"
```
Comment (dpatsora) [inline templates/compose/superset.yaml.erb]: Fix `loqadbalancer`

File: templates/compose/vendor.yaml.erb [hunk 6]
```diff
@@ -15,6 +15,7 @@ services:
         yarn
         yarn start
     labels:
-      traefik.enable: true
-      traefik.frontend.rule: "PathPrefix:/;Host:<%= @config['app']['subdomain'] %>.<%= @config['app']['domain'] %>"
-      traefik.port: 3000
+      - "traefik.http.routers.frontend.rule=Host(`<%= @config['app']['subdomain'] %>.<%= @config['app']['domain'] %>`) && PathPrefix(`/`)"
```
Comment (dpatsora) [inline templates/compose/vendor.yaml.erb]: frontend-<% @name %>

File: templates/compose/vendor.yaml.erb [hunk 7]
```diff
@@ -15,6 +15,7 @@ services:
...
+      - "traefik.http.services.frontend.loqadbalancer.server.port=3000"
```
Comment (dpatsora) [inline templates/compose/vendor.yaml.erb]: Fix `loqadbalancer`

File: templates/compose/vendor.yaml.erb [hunk 8]
```diff
@@ -1,5 +1,7 @@
 version: '3.6'
 
+<%- @name = @config['app']['name'] -%>
```
Comment (vshatravenko) [inline templates/compose/vendor.yaml.erb]: You could've just declared this var in the Rakefile instead of copying it to every template

Comment (akulakovaa) [reply templates/compose/vendor.yaml.erb]: okay

File: templates/compose/proxy.yaml.erb [hunk 9]
```diff
@@ -3,13 +3,25 @@ version: '3.6'
...
+      - "--log.level=DEBUG"
+      - "--api.insecure=true"
+      - "--providers.docker=true"
+      - "--providers.docker.exposedbydefault=false"
+      - "--entrypoints.web.address=:80"
+      - "--entrypoints.websecure.address=:443"
+      <%- if @config['ssl']['enabled'] -%>
+      - "--certificatesresolvers.myresolver.acme.httpchallenge=true"
+      - "--certificatesresolvers.myresolver.acme.httpchallenge.entrypoint=web"
+      - "--certificatesresolvers.myresolver.acme.email=support@openware.com"
```
Comment (vshatravenko) [inline templates/compose/proxy.yaml.erb]: Please add email interpolation from @config here

**Pull request merged**